onepassword/cli/op.go

```diff
@@ -127,12 +139,26 @@ func (op *OP) UpdateItem(ctx context.Context, item *onepassword.Item, vaultUuid
 }
 
 func (op *OP) DeleteItem(ctx context.Context, item *onepassword.Item, vaultUuid string) error {
+	_, err := op.withRetry(func() (*onepassword.Item, error) {
+		return op.delete(ctx, item, vaultUuid)
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (op *OP) delete(ctx context.Context, item *onepassword.Item, vaultUuid string) (*onepassword.Item, error) {
```

Comment:
We should do that to be able to use `delete` with `WithRetry` func. If you take a look at `WithRetry` implementation you'll notice that the `action` function is supposed to return 2 things (item and error). If `delete` returns only `error` it will brake the flow.

So, there were 2 options:
1. Either implement another one `WithRetry` that's used only with `delete`
2. Change the delete signature to return item and error. (and that option I chose 😄 )